app.rb

```diff
@@ -0,0 +1,169 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'classroom'
+require_relative 'specialization'
+require 'date'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    @classrooms = []
+    @specializations = []
+  end
+
+  def list_books
+    puts 'List of All Books:'
+    @books.each { |book| puts " Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_people
+    puts 'List of People:'
+    @people.each do |person|
+      case person
+      when Teacher
+        puts "[Teacher] Name:#{person.name} ID: #{person.id}  Age:#{person.age}"
+      when Student
+        puts "[Student] Name:#{person.name} ID:#{person.id} Age:#{person.age}"
+      else
+        puts "Unknown Type: Name: #{person.name} ID:#{person.id} Age:#{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a teacher (1) or a student (2)? [Input the number]'
+    type_option = gets.chomp.to_i
+    case type_option
+    when 1 then create_teacher
+    when 2 then create_student
+    else
+      puts 'Invalid option'
+    end
+  end
+
+  def create_teacher
+    puts 'Enter age:'
+    age = gets.chomp.to_i
+
+    puts 'Enter name for the teacher:'
+    name = gets.chomp
+
+    puts 'Specialization:'
+    specialization_label = gets.chomp
+
+    specialization = find_or_create_specialization(specialization_label)
+
+    teacher = Teacher.new(name: name, age: age, specialization: specialization)
+    @people << teacher
+    puts 'Person created successfully!'
+  end
+
+  def find_or_create_specialization(label)
+    specialization = @specializations.find { |s| s.label == label }
+    specialization ||= Specialization.new(label)
```

Comment:
- [ ]  🚀 Great work on the app! I tried running it and spotted an Argument error. 🐞

![image](https://github.com/Zel-hub7/school-library/assets/98453979/cbfee037-7191-4ae0-b078-e7756cf1c1a7)


- [ ] 🕵️‍♂️ After some debugging, I noticed that there's a mismatch in how arguments are passed to the `Person` class for teachers. 😬 When creating a teacher, you're using keyword arguments, while the `Person` class expects positional arguments.

- [ ] 🛠️ To resolve this, let's maintain consistency and use positional arguments for the `Teacher` class. Update the `Teacher` class's `initialize` method like this:

### Example

```ruby
def initialize(specialization, age, name = 'Unknown', parent_permission: true)
  super(age, name, parent_permission: parent_permission)
  @specialization = specialization
end

```

- [ ]  📝 Also, make sure to update the method that creates a teacher to match the new argument order:
```
teacher = Teacher.new(specialization, age, name)

```

- [ ] 🧑‍🎓 Let's apply the same approach to resolve the error when creating a student. 

#### 👏 Keep up the good work! 🌟